pkg/config/types/gen_viper/generate.go

```diff
@@ -57,21 +57,24 @@ func generateSetDefaults(t reflect.Type, prefix string, path []string, value ref
 			// Inserting the viper.SetDefault for the nested struct path.
 			constantNameForStruct := strings.ReplaceAll(newPrefix, ".", "")
 			defaultValueForStruct := "cfg." + strings.Join(newPath, ".")
-			fmt.Fprintf(writer, "\tviper.%s(%s, %s)\n", methodName, constantNameForStruct, defaultValueForStruct)
+			fmt.Fprintf(writer, "\tp.Viper.%s(%s, %s)\n", methodName, constantNameForStruct, defaultValueForStruct)
 
 			generateSetDefaults(field.Type, newPrefix, newPath, fieldValue, writer, methodName)
 		} else {
 			constantName := strings.ReplaceAll(newPrefix, ".", "")
 			constantPath := strings.Join(newPath, ".")
 			defaultValue := "cfg." + constantPath
 
-			// Check if the field implements String() method
-			stringerMethod := fieldValue.MethodByName("String")
-			if stringerMethod.IsValid() {
-				defaultValue = "cfg." + constantPath + ".String()"
-			}
+			/*
+				// Check if the field implements String() method
+				stringerMethod := fieldValue.MethodByName("String")
+				if stringerMethod.IsValid() {
+					defaultValue = "cfg." + constantPath + ".String()"
+				}
+
+			*/
```

Comment:
I realise it's commented out but what about ..

```go
_, ok := interface{}(valueField).(fmt.Stringer)
if ok { ... }
```

or there's reflect.TypeOf().Implements()

Reply:
Yeah that would be better if this code was sticking around, but deleting the commented stuff in commit to follow.